### PR TITLE
Allow animation of flashing dots for fronts

### PR DESCRIPTION
**Pull request by OllysCoding**
## What does this change?

Allows the animation of the flashing dot for liveblogs on DCR fronts:

image
(this is a still image)

This wasn't previously working as we only 'animate' the pulsing dot on the client. Although the animation itself could be server-side-rendered, we need to check the users accessibility settings before attempting to show an animation. Cards had previously only been used in places where the entire card is already hydrated (onwards), so the pulsing dot was being animated that way. In fronts cards are server side rendered, so they weren't checking the users preferences & enabling the animation.

This PR adds a new `AnimatePulsingDots.importable.tsx` island which can be applied at a page level to animate pulsing dots. This means we only have a single island even when there are multiple pulsing dots, and I've fully documented the relationship in comments between the two components.

#### Is there an alternative?

The alternative would be too add some way for cards to be aware of whether they will be hydrated (e.g a prop) and drilling it down to `Kicker.tsx`, where it could optionally wrap the existing `PulsingDots.importable.tsx` in an island. 

I decided against this that relationship would leave room for human error when creating / working on cards, introducing a new, implicit problem to consider when trying to use Islands.

### CODE REVIEW

**Comment by Georges-GNM** [comment]
In case this is helpful for posterity, a moving image showing it works (and respects the accessibility setting) :)

https://user-images.githubusercontent.com/102960844/214866045-eca2ffc7-de5b-4ada-849c-c26cd492966d.mov

**Review by Georges-GNM — APPROVED** [review]
Looks good!

File: dotcom-rendering/src/web/components/PulsingDot.importable.tsx [hunk 1]
```diff
@@ -49,7 +53,34 @@ export const PulsingDot = ({ colour }: Props) => {
...
+		setHydrated(true);
 	}, []);
 
-	return <span css={[dotStyles(colour), shouldFlash && animate]} />;
+	return (
+		<span
+			css={[dotStyles(colour), animate]}
+			data-flashing-dot-hydrated={hydrated}
+			/**
+			 * How is the pulsing dot animated?
```
Comment (bryophyta) [inline dotcom-rendering/src/web/components/PulsingDot.importable.tsx:67]: v helpful!

**Comment by OllysCoding** [comment]
> @AshCorr I'm slightly confused why PulsingDot.importable.tsx still needs to be an island instead of solely relying on AnimatePulsingDot.importable.tsx? Is it so that we don't need to use AnimatePulsingDot.importable.tsx on Liveblogs?

Yes, but maybe this isn't the best way of doing it? It leaves 3 different ways of doing the hydration which seems like it could be confusing. I think I'll edit this now :)

**Pull request merged**